Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -0,0 +1,250 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file KrBoxClusterFinder3D.cpp
+/// \brief Class source code for Krypton and X-ray events
+/// \author Philip Hauer <hauer@hiskp.uni-bonn.de>
+
+#include "TPCReconstruction/KrBoxClusterFinder.h"
+#include <iostream>
+
+using namespace o2::tpc;
+
+// Constructor:
+// This function creates a three dimensional vector (Pad,Row,Time) which is
+// filled with the charges
+KrBoxClusterFinder::KrBoxClusterFinder(std::vector<o2::tpc::Digit>* eventSector)
+{
+  std::vector<o2::tpc::Digit>::iterator cl = eventSector->begin();
+  std::vector<int> Rows, Pads, Times;
+
+  if (eventSector->size() == 0) {
+    // prevents a segementation fault if the envent contains no data
+    // segementation fault would occure later when trying to dereference the
+    // max/min pointer (which are nullptr if data is empty) empty events should
+    // be catched in the main function, hence, this "if" is no longer necessary
+    std::cout << "WARNING SECTOR SIZE (AMOUNT OF DATA POINTS) IN CURRENT RUN IS 0" << std::endl;
+    std::cout << "mMapOfAllDigits WITH 0's IS GENERATED" << std::endl;

```

Comment:
Please change all cout to logger output (everywhere in the code)
add `#include "Framework/Logger.h" on the top, then use
```c++
LOGP(warning, "Sector size ...");
```
`LOGP` uses libfmt formatting, so you can do something like:
```c++
LOGP(info, "Number of digits in event: {}", eventSector->size());
```
Just check libfmt: https://fmt.dev/latest/

you also have `error`, `info`, `fatal`, `debug`. Where `fatal` will kill the program.

Reply:
Done.